.github/workflows/tests.yml

```diff
@@ -0,0 +1,76 @@
+name: Tests
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  tests:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ] #, macos-latest, windows-latest ]
+        php-version: [ "7.2", "7.3", "7.4" ]
+
+    # https://docs.github.com/en/free-pro-team@latest/actions/guides/about-service-containers
+    services:
+      cratedb:
+        image: crate:latest
+        ports:
+          - 4200:4200
+
+    # https://github.com/marketplace/actions/setup-php-action
+    name: PHP ${{ matrix.php-version }} on OS ${{ matrix.os }}
+    steps:
+
+      - name: Acquire sources
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          # Select PHPUnit version suitable for PHP 7.2.
+          tools: composer, phpunit:8.5.13
+
+      # Cache Composer Dependencies
+      # https://github.com/marketplace/actions/setup-php-action#cache-composer-dependencies
+      - name: Get composer cache directory
+        id: composer-cache-dir
+        run: echo "::set-output name=dir::$(composer config cache-dir)"
+      - name: Cache dependencies
+        id: composer-cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.composer-cache-dir.outputs.dir }}
+          key: os=${{ matrix.os }}-composer-${{ hashFiles('**/composer.lock') }}
+
+      - name: Install dependencies
+        run: |
+          
+          # Library dependencies
+          # Remark: --prefer-source is needed for "Doctrine\Tests\DBAL\Platforms\AbstractPlatformTestCase"
+          composer install --prefer-source
+
+          # Coverage tools
+          composer global require php-coveralls/php-coveralls
+          composer global require scrutinizer/ocular
+
+      - name: Run tests
+        run: phpunit --coverage-clover build/logs/clover.xml
+
+      - name: Upload coverage
+        continue-on-error: true
+        run: |
+
+          # Run all commands, but don't mask exit code
+          exitcode=0; on_error() { exitcode=1; }; trap on_error ERR
+
+          # Upload to Coveralls
+          php-coveralls --coverage_clover=build/logs/clover.xml -v
+
+          # Upload to Scrutinizer
+          ocular code-coverage:upload --format=php-clover build/logs/clover.xml
+
+          exit $exitcode
```

Comment:
We will yet have to provide authentication tokens for Coveralls and Scrutinizer before integrating this.

Reply:
It works.

- https://coveralls.io/github/crate/crate-dbal
- https://scrutinizer-ci.com/g/crate/crate-dbal/